### PR TITLE
[stdlib] Rename `VectorProtocol.*` to `VectorProtocol.scaled(by:)`.

### PR DESCRIPTION
**Pull request by dan-zheng**
Rename `VectorProtocol` scalar multiplication.
- `VectorProtocol.*` -> `VectorProtocol.scaled(by:)`
- `VectorProtocol.*=` -> `VectorProtocol.scale(by:)`

Update `VectorProtocol` derived conformances.

### CODE REVIEW

**Comment by dan-zheng** [comment]
Note: `swift-apis` compilation currently fails for `BatchNorm` and `LayerNorm` in `tensorflow-swift-apis/Sources/TensorFlow/Layers/Normalization.swift`.

Minimal reproducer:
```swift
import TensorFlow

public struct LayerNorm<Scalar: TensorFlowFloatingPoint>: Layer {
    public var offset: Tensor<Scalar>
    public var scale: Tensor<Scalar>
    @noDerivative public let axis: Int
    @noDerivative public let epsilon: Tensor<Scalar>

    @differentiable
    public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
        return input
    }
}
```

```
<unknown>:0: error: cannot convert value of type 'WritableKeyPath<LayerNorm<Scalar>.AllDifferentiableVariables, Tensor<Scalar>>' to expected element type 'PartialKeyPath<LayerNorm<_>.AllDifferentiableVariables>'
```

This is weird because the error appears to be related to `KeyPathIterable` derived conformances, which are unchanged by this PR.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Review by rxwei — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please clean test tensorflow
